### DISCUSSION TITLE
Allow customization of ElementQuery statusCondition

### DISCUSSION THREAD

**Discussion by lenvanessen** [craftcms/cms#9047] [Ideas]
### Description
Following #7694, we're investigating the possibility of overriding Crafts default postDate field, so that we can have a different date for each site and allow them to publish on a different schedule, while remaining Craft awesome propagation functionality.

The current approach we'd like to take, is to override the `statusCondition` method on the `ElementQuery` class. By adding a couple of extra lines we can make this work for our use case. However, currently I don't see a way to alter this method without overloading the entire class in Yii's di. 

Would it be possible to think of a way to customize the conditions that are set around the statuses?

**Comment by Anubarak** [comment]
You can create a behaviour and attach it to the EntryQuery and use the `beforePrepare` event to include your custom check while unsetting the status value and re-set it in the `afterPrepare` (otherwise you won't be able to execute queries twice)

**Comment by lenvanessen** [comment]
Thanks cap, I’ll give it a shot!

**Comment by Anubarak** [comment]
Just some kind of example how to use the behaviour in general (don't have a better example right now) https://gist.github.com/Anubarak/503d86263da1482aa545645071050c68

This one includes a custom query parameter in order to show only entries in the cp that belong to a user via custom fields... Just change the logic in order to suit your use case.
